Json <-> Dynamic Obj converter

Freymaurer opened the pull request:
- Added to- and from-json converter function.
- Added unit tests (12).
- Run unit tests (51/51 ✔️ ).

kMutagene commented on src/DynamicObj/DynamicObj.fs:
```diff
@@ -135,4 +136,129 @@ type DynamicObj internal (dict:Dictionary<string, obj>) =
+
+module DynamicObj =
+
+    let toJson (dynObj:DynamicObj) = JsonConvert.SerializeObject(dynObj, new DynamicObjConverter())
```
please add inline XML docs (`<summary>` and `<param>` tags) This is something we should do for all public APIs in the future.

kMutagene commented on src/DynamicObj/DynamicObj.fs:
```diff
@@ -135,4 +136,129 @@ type DynamicObj internal (dict:Dictionary<string, obj>) =
+    let toJson (dynObj:DynamicObj) = JsonConvert.SerializeObject(dynObj, new DynamicObjConverter())
+
+    /// Read json string to DynamicObj. If json string has an array as root element the array will be set as value with "root" as key.
+    let ofJson (jsonSource:string) = JsonConvert.DeserializeObject<DynamicObj>(jsonSource, new DynamicObjConverter())
```
please add inline XML docs

kMutagene commented on tests/UnitTests/Tests.fs:
```diff
@@ -208,4 +208,118 @@ let ``test print for issue 14``() =
+let private minifyJson(json:string) = 
+    json
+        .Replace(" ","")
+        // if i add this line, my tests break
```
You might want to implement the tests using the JSON as embedded resources to prevent newline problems. see https://github.com/fslaborg/FSharp.Stats/commit/23b49a95a8167829643cf743d296a14b8081907c

Freymaurer commented on src/DynamicObj/DynamicObj.fs:
```diff
@@ -1,8 +1,9 @@
-﻿namespace DynamicObj
+﻿namespace rec DynamicObj
```
This was done because the `DynamicObjConverter` needs `DynamicObj`, while in turn `DynamicObj.ToJson()` needs the `DynamicObjConverter`.

Freymaurer commented:
> Sorry for ghosting this PR @Freymaurer. I think it was bad timing as we moved to a transpilable version of this lib. However, i think we can use your JToken unboxing logic to create a out-of-the-box deserializer via `JsonExtensionData` (https://www.newtonsoft.com/json/help/html/SerializationAttributes.htm#JsonExtensionDataAttribute). I will look into it now that we fixed #37 

No worries i did not even remember this PR anymore

HLWeil commented:
Probably outdated and can be closed? @Freymaurer @kMutagene
